app/src/main/java/org/simple/clinic/registration/facility/RegistrationFacilitySelectionScreen.kt

```diff
@@ -109,20 +109,21 @@ class RegistrationFacilitySelectionScreen(context: Context, attrs: AttributeSet)
           .map(::RegistrationFacilityClicked)
 
   private fun activityResultStream(): Observable<UiEvent> {
-    val screenDestroys: Observable<ScreenDestroyed> = detaches().map { ScreenDestroyed() }
-
     return screenRouter
         .streamScreenResults()
         .ofType<ActivityResult>()
         .filter { it.requestCode == CONFIRM_FACILITY_SHEET && it.succeeded() && it.data != null }
-        .takeUntil(screenDestroys)
+        .takeUntil(onDestroy())
         .map { it.data!! }
         .map { intent ->
           val confirmedFacilityUuid = ConfirmFacilitySheet.confirmedFacilityUuid(intent)
           RegistrationFacilityConfirmed(confirmedFacilityUuid)
         }
   }
 
+  private fun onDestroy() =
+      detaches().map { ScreenDestroyed() }
```

Comment:
Since this stream is required only in `onFinishInflate()`, we can make this stream a variable instead of a separate method.

Reply:
It's being called in `onFinishInflate()` & `registrationFacilityConfirmations()`

Reply:
Yeah, but `registrationFacilityConfirmations` is itself called inside `onFinishInflate`, so we can create a variable and pass it as a parameter.

Reply:
Done. PTAL